lib/vicadmin/validate.go

```diff
@@ -138,18 +139,57 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 	return v
 }
 
+type dsList []mo.Datastore
+
+func (d dsList) Len() int { return len(d) }
+func (d dsList) Swap(i, j int) { d[i], d[j] = d[j], d[i] }
+func (d dsList) Less(i, j int) bool { return d[i].Name < d[j].Name }
+
 func (v *Validator) QueryDatastore(ctx context.Context, vch *config.VirtualContainerHostConfigSpec, sess *session.Session) {
-	var ds []mo.Datastore
-	refs := []types.ManagedObjectReference{sess.Datastore.Reference()}
+	var dataStores dsList
+	dsNames := []string{}
+
+	for _, url := range vch.ImageStores {
+		dsNames = append(dsNames, url.Host)
+	}
+
+	for _, url := range vch.VolumeLocations {
+		dsNames = append(dsNames, url.Host)
+	}
+
+	for _, url := range vch.ContainerStores {
+		dsNames = append(dsNames, url.Host)
+	}
+
+	sort.Strings(dsNames)
+	lastDs := ""
+
+	refs := []types.ManagedObjectReference{}
+	for _, dsName := range dsNames {
+		if lastDs == dsName {
+			continue
+		}
+		lastDs = dsName
+		ds, err := sess.Finder.DatastoreOrDefault(ctx, dsName)
```

Comment:
Should use Finder.Datastore method. You don't need to use default datastore if the dsName is ""